python/feathub/processors/local/sliding_window_utils.py

```diff
@@ -0,0 +1,350 @@
+#  Copyright 2022 The Feathub Authors
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from datetime import timedelta, datetime, tzinfo, timezone
+from typing import Optional, Sequence, Type, Any, Tuple, Dict, List, Callable
+
+import pandas as pd
+
+from feathub.common import utils
+from feathub.common.exceptions import FeathubException
+from feathub.common.types import to_numpy_dtype
+from feathub.dsl.ast import ExprAST
+from feathub.dsl.expr_parser import ExprParser
+from feathub.feature_views.feature import Feature
+from feathub.feature_views.sliding_feature_view import SlidingFeatureView
+from feathub.feature_views.transforms.agg_func import AggFunc
+from feathub.feature_views.transforms.sliding_window_transform import (
+    SlidingWindowTransform,
+)
+from feathub.processors.local.aggregation_utils import AGG_FUNCTIONS
+from feathub.processors.local.ast_evaluator.local_ast_evaluator import LocalAstEvaluator
+from feathub.processors.local.type_utils import cast_dataframe_dtype
+
+
+class AggregationFieldDescriptor:
```

Comment:
Can we unify classes like this across all processors? Maybe add a TODO for future work.